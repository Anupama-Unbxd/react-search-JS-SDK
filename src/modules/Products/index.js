import React from 'react';
import AppContext, { AppContextConsumer } from '../../common/context'
import { ProductContextProvider } from './context'

import PropTypes from 'prop-types';

import ViewTypes from './ViewTypes';
import ProductsView from './ProductsView';
import { conditionalRenderer } from '../../common/utils';

class Products extends React.PureComponent {

    ProductContext = React.createContext();

    static contextType = AppContext;

    static ViewTypes = ViewTypes;
    static ProductsView = ProductsView;

    onViewToggle = (event) => {
        this.setState({ isGrid: event.target.id === 'Unbx-grid' })
    }

    constructor(props) {
        super(props);

        //Set all data that is required in state so that it can be passed down to children
        const { ZeroResultsTemplate, per_row, per_page,
            infiniteScroll, loadOnClick, fieldMap, variantMap } = this.props

        this.state = {
            isGrid: true,
            onViewToggle: this.onViewToggle,
            ZeroResultsTemplate,
            per_row,
            per_page,
            infiniteScroll,
            loadOnClick,
            fieldMap,
            variantMap
        }
    }

    componentDidMount() {

        const { helpers: { setProductConfiguration } } = this.context;

        const { per_page, fieldMap, variants,
            variantCount, variantMap } = this.props

        //generate required fields here based on fieldMap and variantFields
        const requiredFields = Object.values(fieldMap);
        const variantRequiredFields = Object.values(variantMap);

        //Set the main config with these local configs
        setProductConfiguration({
            per_page, requiredFields, variants, variantCount, variantRequiredFields
        });
    }

    render() {

        const { per_page } = this.state;

        return (<AppContextConsumer>
            {({ unbxdCore }) => {

                const getSearchResults = unbxdCore.getSearchResults.bind(unbxdCore);
                //Wrap this in a function so that there is only one function call
                const setPageStart = unbxdCore.setPageStart.bind(unbxdCore);
                const getPaginationInfo = unbxdCore.getPaginationInfo.bind(unbxdCore);
                const getResults = unbxdCore.getResults.bind(unbxdCore);

                //Get next page method
                const getNextPage = () => {

                    //Get the next batch by jumping the current per_page
                    const { currentPage } = getPaginationInfo();
                    const newStart = currentPage === 0 ? per_page : currentPage + per_page;
                    setPageStart(newStart);
                    getResults();
                }

                const unbxdProductCardClickHandler = () => {
                    console.log("hello from unbxdProductCardClickHandler")
                }

                //onClick for products
                //We also need to pass infinite scroll method
                const getProductsProp = ({ onClick, ...props }) => ({
                    onClick: () => {
                        unbxdProductCardClickHandler();
                        onClick();
                    },
                    getSearchResults,
                    getNextPage,
                    ...props
                })

                const DefaultRender = <React.Fragment>
                    <ViewTypes />
                    <ProductsView />
                </React.Fragment>

                //We will be using components here which can also be used as compound components
                //We need to send prop getters for render props. mainly onclick
                return (<ProductContextProvider value={{ ...this.state, getSearchResults, getNextPage, unbxdProductCardClickHandler }}>
                    {conditionalRenderer(this.props.children, {
                        state: this.state,
                        getProductsProp,
                    }, DefaultRender)}
                </ProductContextProvider>)
            }
            }
        </AppContextConsumer >)
    }
}

Products.defaultProps = {
    per_row: 5,
    per_page: 10,
    //requiredFields: ["imageUrl", "title", "price"],
    fieldMap: { productName: "title", uniqueId: "uniqueId", imageUrl: "imageUrl" },
    ZeroResultsTemplate: false,
    infiniteScroll: false,
    loadOnClick: false,
    variants: false,
}

Products.propTypes = {
    per_row: PropTypes.number,
    per_page: PropTypes.number,
    fieldMap: PropTypes.object.isRequired,
    ZeroResultsTemplate: PropTypes.oneOfType([PropTypes.element, PropTypes.bool, PropTypes.func]),
    infiniteScroll: PropTypes.bool,
    loadOnClick: PropTypes.bool,
    variants: PropTypes.bool,
    variantCount: PropTypes.number,
    variantMap: PropTypes.object,
    variantRequiredFields: PropTypes.arrayOf(PropTypes.string),
}

export default Products;
