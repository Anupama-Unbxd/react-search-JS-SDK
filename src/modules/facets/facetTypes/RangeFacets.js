import React from 'react';


const RangeFacets = () => {
    return (<div>Hello from RangeFacets</div>)
}

export default RangeFacets;
