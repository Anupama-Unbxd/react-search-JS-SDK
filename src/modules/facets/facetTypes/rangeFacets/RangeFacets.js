import React from 'react';
import PropTypes from 'prop-types';

import { FacetsContextConsumer } from '../../context';
import GenerateFacets from './GenerateFacets';

const RangeFacets = () => {

    return (<FacetsContextConsumer>
        {({ data, helpers }) => {

            const { rangeFacets, selectedFacets, isApplyFilters, selectedRangeFacets } = data;
            const { addRangeFacet, applyRangeFacet, clearARangeFacet, trackActions } = helpers;


            return (<GenerateFacets
                rangeFacets={rangeFacets}
                selectedFacets={selectedFacets}
                isApplyFilters={isApplyFilters}
                selectedRangeFacets={selectedRangeFacets}
                addRangeFacet={addRangeFacet}
                applyRangeFacet={applyRangeFacet}
                clearARangeFacet={clearARangeFacet}
                trackActions={trackActions} />)
        }
        }</FacetsContextConsumer>)

}



export default RangeFacets;
