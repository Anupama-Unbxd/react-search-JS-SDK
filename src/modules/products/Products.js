import React from 'react';
import PropTypes from 'prop-types';

import AppContext from '../../common/context'
import { ProductContextProvider } from './context'
import ViewTypes from './viewTypes/ViewTypes';
import ProductsView from './productsView/ProductsView';
import { conditionalRenderer, hasUnbxdSearchWrapperContext } from '../../common/utils';
import { Loader as defaultLoader } from '../../components'
import { getProductViewType } from './utils';
import { trackProductClick } from '../analytics';


/**
 * Component to render the search products. 
 * Products supports Product View types `LIST` and `GRID` out of the box. 
 * Products also manages the pagination options for the search results here.
 */
export class Products extends React.PureComponent {

    constructor(props) {
        super(props);

        const { productViewTypes } = this.props

        //Lets fallback to grid view
        const productViewType = getProductViewType(productViewTypes)[0];

        this.state = {
            productViewType,
        }
    }

    componentDidMount() {

        if (this.context === undefined) {
            hasUnbxdSearchWrapperContext(Products.displayName);
        }

        const { helpers: { setProductConfiguration } } = this.context;

        const { pageSize,
            attributesMap,
            showVariants,
            variantsCount,
            variantAttributesMap,
            paginationType,
            groupBy } = this.props;

        //generate required fields here based on attributesMap and variantAttributesMap
        const requiredFields = Object.values(attributesMap);
        const variantRequiredFields = Object.values(variantAttributesMap);

        //Set the main config
        //Do not set pageSize if pagination type is FIXED_SCROLLING
        setProductConfiguration({
            pageSize,
            requiredFields,
            showVariants,
            variantsCount,
            variantRequiredFields,
            paginationType,
            groupBy
        });
    }

    getProductProps() {

        if (this.context === undefined) {
            hasUnbxdSearchWrapperContext(Products.displayName);
        }

        const { unbxdCore, unbxdCoreStatus, helpers: { trackActions } } = this.context;
        const { ZeroResultsComponent,
            perRow,
            pageSize,
            paginationType,
            productViewTypes,
            productViewDisplayType,
            heightDiffToTriggerNextPage,
            LoaderComponent,
            showLoader,
            onProductClick: onProductClickCB,
            LoadMoreComponent,
            onZeroResults,
            ProductItemComponent,
            ProductsViewItemComponent,
            productIdAttribute,
            attributesMap,
            showVariants,
            groupBy,
            variantAttributesMap,
            showSwatches,
            swatchAttributes,
            SwatchItemComponent } = this.props;

        const getSearchResults = unbxdCore.getSearchResults.bind(unbxdCore);
        const setPageStart = unbxdCore.setPageStart.bind(unbxdCore);
        const getPaginationInfo = unbxdCore.getPaginationInfo.bind(unbxdCore);
        const getResults = unbxdCore.getResults.bind(unbxdCore);
        const getProductByPropValue = unbxdCore.getProductByPropValue.bind(unbxdCore);

        const query = unbxdCore.getSearchQuery() || "";

        const onViewToggle = (event) => {

            const productViewType = event.target.dataset.viewtype || event.target.value;;

            this.setState({ productViewType });
        }

        const getNextPage = () => {

            const { currentPage } = getPaginationInfo();
            const newStart = currentPage === 0 ? pageSize : (currentPage * pageSize);

            setPageStart(newStart);
            getResults();
        }

        const onProductClick = (event) => {

            const productUniqueId = event.target.dataset.uniqueid;
            if (productUniqueId) {
                const prank = event.target.dataset.prank;
                const clickedProduct = getProductByPropValue(productIdAttribute, productUniqueId);

                if (onProductClickCB) {

                    if (onProductClickCB(clickedProduct)) {

                        trackProductClick(clickedProduct[productIdAttribute], prank);
                    } else {

                        event.preventDefault();
                    }
                } else {

                    trackProductClick(clickedProduct[productIdAttribute], prank);
                }

            }

        }

        const getOnProductsClickProps = ({ onClick, ...props }) => ({
            onClick: () => {
                onProductClick();
                onClick();
            },
            ...props
        })

        const data = {
            unbxdCoreStatus,
            perRow,
            paginationType,
            attributesMap,
            variantAttributesMap,
            productViewTypes,
            heightDiffToTriggerNextPage,
            showVariants,
            showLoader,
            productViewDisplayType,
            showSwatches,
            swatchAttributes,
            groupBy,
            query,
            productIdAttribute,
            ...this.state
        };
        const helpers = {
            ZeroResultsComponent,
            onViewToggle,
            getOnProductsClickProps,
            getNextPage,
            onProductClick,
            getSearchResults,
            getNextPage,
            LoaderComponent,
            onZeroResults,
            ProductItemComponent,
            ProductsViewItemComponent,
            SwatchItemComponent,
            LoadMoreComponent
        }

        return { data, helpers }

    }

    render() {

        const DefaultRender = <React.Fragment>
            <ViewTypes />
            <ProductsView />
        </React.Fragment>;
        const LoaderRender = null;


        return (<ProductContextProvider value={this.getProductProps()}>
            {conditionalRenderer(this.props.children, this.getProductProps(), DefaultRender, LoaderRender)}
        </ProductContextProvider>)
    }
}

Products.contextType = AppContext;
Products.ViewTypes = ViewTypes;
Products.ProductsView = ProductsView;
Products.displayName = "Products";

Products.defaultProps = {
    perRow: 4,
    pageSize: 10,
    productViewTypes: ["GRID"],
    attributesMap: {},
    variantAttributesMap: {},
    paginationType: 'FIXED_PAGINATION',
    heightDiffToTriggerNextPage: 100,
    showVariants: false,
    LoaderComponent: defaultLoader,
    showLoader: false,
    productViewDisplayType: "DROPDOWN",
    productIdAttribute: 'uniqueId',
}

Products.propTypes = {
    /**
    * Number of products to be shown per row.
    */
    perRow: PropTypes.number,
    /**
    * Number of products to be loaded on a page.
    */
    pageSize: PropTypes.number,
    /**
    * Required pagination type. Possible options are `INFINITE_SCROLL`, `CLICK_N_SCROLL` and `FIXED_PAGINATION`.
    */
    paginationType: PropTypes.string,
    /**
     * Height difference to trigger for next page in case of paginationType `INFINITE_SCROLL`.
     */
    heightDiffToTriggerNextPage: PropTypes.number,
    /**
     * Custom load more component for CLICK_N_SCROLL
     */
    LoadMoreComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
    /**
    * Required ProductViewType.Possible options are`GRID` and `LIST`.
    */
    productViewTypes: PropTypes.arrayOf(PropTypes.string),
    /**
    * Required ProductViewType Display type.Possible options are`LIST` and `DROPDOWN`.
    */
    productViewDisplayType: PropTypes.string,
    /**
    * Custom `LIST` item component for product views.
    */
    ProductsViewItemComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
    /**
    * Unique attribute of the product
    */
    productIdAttribute: PropTypes.string,
    /**
    * Mapping of catalog Product fields to SDK Product fields.
    */
    attributesMap: PropTypes.object.isRequired,
    /**
    * Show variants to a product.
    */
    showVariants: PropTypes.bool,
    /**
    * Number of variants to fetch.
    */
    variantsCount: PropTypes.number,
    /**
    * Mapping of catalog Product variant fields to SDK Product variant fields.
    */
    variantAttributesMap: PropTypes.object,
    /**
     * Should loader be shown
     */
    showLoader: PropTypes.bool,
    /**
    * Custom loader component
    */
    LoaderComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
    /**
    * Callback function triggered on click of a product.
    */
    onProductClick: PropTypes.func,
    /**
    * Callback function triggered on zero results.
    */
    onZeroResults: PropTypes.func,
    /**
    * Display swatches
    */
    showSwatches: PropTypes.bool,
    /**
    * Group variants by the attribute
    */
    groupBy: PropTypes.string,
    /**
    * Swatch attributes that change on click on of the swatch
    */
    swatchAttributes: PropTypes.object,
    /**
    * Custom swatch component
    */
    SwatchItemComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
    /**
    * Custom product item component
    */
    ProductItemComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
    /**      
    *  Component to be shown in case of zero results.
    */
    ZeroResultsComponent: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
}

export default Products;
