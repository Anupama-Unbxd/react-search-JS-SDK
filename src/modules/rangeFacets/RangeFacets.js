import React from 'react';
import PropTypes from 'prop-types';

import { AppContextConsumer } from '../../common/context';
import { hasUnbxdSearchWrapperContext } from '../../common/utils';
import RangeFacetsContainer from './RangeFacetsContainer';

/**
 * Component to manage range facets.
 */
const RangeFacets = (props) => {
  return (
    <AppContextConsumer>
      {(appState) => {
        if (appState === undefined) {
          hasUnbxdSearchWrapperContext(RangeFacets.displayName);
        }

        const {
          unbxdCore,
          unbxdCoreStatus,
          helpers,
          unbxdState,
          priceUnit
        } = appState;
        const { enableApplyFilters } = unbxdState;

        return (
          <RangeFacetsContainer
            unbxdCore={unbxdCore}
            unbxdCoreStatus={unbxdCoreStatus}
            helpers={helpers}
            enableApplyFilters={enableApplyFilters}
            priceUnit={priceUnit}
            {...props}
          />
        );
      }}
    </AppContextConsumer>
  );
};

RangeFacets.displayName = 'RangeFacets';

RangeFacets.defaultProps = {
  displayType: 'SLIDER',
  collapsible: true
};

RangeFacets.propTypes = {
  /**
   * display type of facet component.
   */
  displayType: PropTypes.string,
  /**
   * Collapse facet values
   */
  collapsible: PropTypes.bool,
  /**
   * Search facet values
   */
  FacetSliderItemComponent: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func
  ]),
  /**
   * Custom facet list item component.
   */
  FacetListItemComponent: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func
  ]),
  /**
   * Label for the component.
   */
  label: PropTypes.node
};

export default RangeFacets;
