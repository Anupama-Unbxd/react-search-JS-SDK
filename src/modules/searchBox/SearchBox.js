import React from 'react';
import PropTypes from 'prop-types';

import AppContext from '../../common/context'
import { SearchBoxContextProvider } from './context'
import SearchInput from './searchInput';
import SearchButton from './searchButton';
import { conditionalRenderer, isContext } from '../../common/utils';

/**
 * Component to set or update the search query.
 */
class SearchBox extends React.Component {

    constructor(props) {
        super(props);

        this.onQueryChange = this.onQueryChange.bind(this);
        this.onClearQuery = this.onClearQuery.bind(this);
        this.handleQuerySubmit = this.handleQuerySubmit.bind(this);

        this.state = { query: '' };
    }


    onQueryChange(event) {
        const query = event.target.value;
        this.setState({ query });
    }

    onClearQuery() {

        const { query } = this.state;
        const { onClear } = this.props;

        if (onClear) {
            onClear(query) && this.setState({ query: '' });
        } else {
            this.setState({ query: '' });
        }
    }

    handleQuerySubmit(event) {
        event.preventDefault();

        const { query } = this.state;
        const { onSubmit } = this.props;
        const { helpers: { setSearchBoxConfiguration, trackActions } } = this.context;

        if (onSubmit) {

            if (onSubmit(query) && query.length) {

                setSearchBoxConfiguration({ query });
                //track for search hit here
                trackActions({ type: 'SEARCH', data: { query } })
            }
        }
        else {

            if (query.length) {

                setSearchBoxConfiguration({ query });
                //track for search hit here
                trackActions({ type: 'SEARCH', data: { query } })
            }
        }


    }

    componentDidMount() {

        if (this.context === undefined) {
            isContext(Products.displayName);
        }
    }

    getSearchBoxProps() {

        const {
            isAutoFocus,
            isClear, } = this.props;

        const data = {
            isAutoFocus,
            isClear,
            ...this.state
        };
        const helpers = {
            onQueryChange: this.onQueryChange,
            handleQuerySubmit: this.handleQuerySubmit,
            onClearQuery: this.onClearQuery,

        };

        return { data, helpers };
    }


    render() {

        const DefaultRender = <form onSubmit={this.handleQuerySubmit} className='UNX-searchbox-container'>
            <SearchInput />
            <SearchButton />
        </form>

        return (<SearchBoxContextProvider value={this.getSearchBoxProps()}>
            {conditionalRenderer(this.props.children, this.getSearchBoxProps(), DefaultRender)}
        </SearchBoxContextProvider>)
    }
}

SearchBox.contextType = AppContext;

SearchBox.SearchInput = SearchInput;
SearchBox.SearchButton = SearchButton;

SearchBox.displayName = "SearchBox";

SearchBox.defaultProps = {
    isAutoFocus: false,
    isClear: false,
    onSubmit: null,
    onClear: null,
}

SearchBox.propTypes = {
    /**
    * Should the searchbox be focused by default.
    */
    isAutoFocus: PropTypes.bool,
    /**
    * Should the searchbox be clearable.
    */
    isClear: PropTypes.bool,
    /**
    * Hook for search query. The function should return `true` if the search is to be triggered, false otherwise.
    */
    onSubmit: PropTypes.func,
    /**
    * Hook for clearing the search query. The function should return `true` if the searchbox is to be cleared, false otherwise.
    */
    onClear: PropTypes.func,
}

export default SearchBox;
