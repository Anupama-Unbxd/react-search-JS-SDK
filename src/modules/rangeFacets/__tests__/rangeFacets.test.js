import renderer from 'react-test-renderer';
import { render, waitFor, fireEvent } from '@testing-library/react';
import React from 'react';
import Products from '../../products/index';
import RangeFacets from '../index';
import UnbxdSearchWrapper from '../../../UnbxdSearchWrapper';
import SearchBox from '../../searchBox';
import { searchResponse } from './mocks/searchMock';
import { facetResponse } from './mocks/searchMock';

export const FacetItemComponent = ({ itemData, onClick, priceUnit }) => {
    const { from, end, facetName, isSelected = false } = itemData;
    const { name: fromName, count, dataId: fromDataId } = from;
    const { name: ToName, dataId: toDataId } = end;

    const handleClick = () => {
        onClick(itemData);
    };

    return (
        <div
            key={`${facetName}_${fromDataId}-${toDataId}`}
            className={`UNX-facet__item ${isSelected ? '-selected' : ''}`}
            onClick={handleClick}
        >
            <div className="-checkbox" />
            <div className="-label">
                {priceUnit}
                {fromName} - {priceUnit}
                {ToName}
            </div>
            <div className="-count">({count})</div>
        </div>
    );
};

// establish API mocking before all tests
beforeAll(() => {
    window.fetch = jest.fn((request) => {
        if (request.includes('[200%20TO%20300]')) {
            return Promise.resolve({
                json: () => Promise.resolve(facetResponse),
            });
        }
        return Promise.resolve({
            json: () => Promise.resolve(searchResponse),
        });
    });
})

const attributesMap = {
    productName: 'title',
    uniqueId: 'uniqueId',
    imageUrl: 'imageUrl',
    price: 'RRP_Price',
    productUrl: 'productUrl'
};


test('Match Snapshot for range facets', async () => {
    const tree = renderer
        .create(
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >
                <RangeFacets />
                <Products attributesMap={attributesMap} />
                <div>
                    <SearchBox defaultSearch="shoes" />
                </div>
            </UnbxdSearchWrapper>

        );
        await waitFor(() => tree.toJSON());
        expect(tree.toJSON()).toMatchSnapshot();
});

test('Test range facet click', async () => {
    const { getByText } = render(
        <>
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >   
                <RangeFacets />
                <Products
                    attributesMap={attributesMap}
                />
                <div>
                    <SearchBox defaultSearch="shoes" />
                </div>
            </UnbxdSearchWrapper>
        </>
    );

    await waitFor(async () => {
        expect(getByText('$ 200 - $ 300 - 60')).toBeInTheDocument();
        fireEvent.click(getByText("$ 200 - $ 300 - 60"));
    });
    await waitFor(() => {
        expect(getByText("Salomon OUTblast TS CSWP Mens Hiking Boots - Black/Black/Black")).toBeInTheDocument();
    });
});

test('Test range facet click with FacetItemComponent', async () => {
    const { getByText } = render(
        <>
            <UnbxdSearchWrapper
                siteKey="wildearthclone-neto-com-au808941566310465"
                apiKey="e6959ae0b643d51b565dc3e01bf41ec1"
            >   
                <RangeFacets facetItemComponent={<FacetItemComponent />}/>
                <Products
                    attributesMap={attributesMap}
                />
                <div>
                    <SearchBox defaultSearch="shoes" />
                </div>
            </UnbxdSearchWrapper>
        </>
    );

    await waitFor(async () => {
        expect(getByText('$200 - $300')).toBeInTheDocument();
        fireEvent.click(getByText("$200 - $300"));
    });
    await waitFor(() => {
        expect(getByText("Salomon OUTblast TS CSWP Mens Hiking Boots - Black/Black/Black")).toBeInTheDocument();
    });
});



