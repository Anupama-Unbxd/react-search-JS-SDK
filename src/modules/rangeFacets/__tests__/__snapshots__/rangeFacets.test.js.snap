// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match Snapshot for range facets 1`] = `
Array [
  <div
    className="UNX-selectedFacets__container"
  >
    <div
      className="UNX-selectedFacets__list"
    >
      <div
        className="UNX-selectedFacets__item"
        onClick={[Function]}
      >
        <span>
          $
           
          200
           - 
          $
           
          300
        </span>
         x
      </div>
    </div>
  </div>,
  <div
    className="UNX-rangefacet__container"
  >
    <div
      className="UNX-facet__element open"
    >
      <div
        className="UNX-facet__headerContainer"
      >
        <div
          className="UNX-facet__header"
        >
          FILTER BY PRICE
          <span
            className="-collapse-icon"
            data-unx_name="price"
            onClick={[Function]}
          />
        </div>
      </div>
      <div
        className="UNX-facet__list "
      >
        <div
          className="UNX-facet__item "
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
        >
          $
           
          0
           - 
          $
           
          100
           - 
          21
        </div>
        <div
          className="UNX-facet__item "
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
        >
          $
           
          100
           - 
          $
           
          200
           - 
          18
        </div>
        <div
          className="UNX-facet__item -selected"
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
        >
          $
           
          200
           - 
          $
           
          300
           - 
          60
        </div>
        <div
          className="UNX-facet__item "
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
        >
          $
           
          300
           - 
          $
           
          400
           - 
          38
        </div>
        <div
          className="UNX-facet__item "
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
        >
          $
           
          400
           - 
          $
           
          600
           - 
          13
        </div>
        <div
          className="UNX-facet__item "
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
        >
          $
           
          600
           - 
          $
           
          700
           - 
          1
        </div>
        <div
          className="UNX-facet__item "
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
        >
          $
           
          700
           - 
          $
           
          900
           - 
          4
        </div>
        <div
          className="UNX-facet__item "
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
        >
          $
           
          900
           - 
          $
           
          1000
           - 
          3
        </div>
        <div
          className="UNX-facet__item "
          data-testid="UNX_rangeFacet__facetItem"
          onClick={[Function]}
        >
          $
           
          1000
           - 
          $
           
          1100
           - 
          1
        </div>
      </div>
      <div
        className="-clear"
        data-unx_facetname="price"
        onClick={[Function]}
      >
        Clear
      </div>
    </div>
  </div>,
  <div>
    <div
      className="UNX-products__container"
    >
      <div
        className="UNX-products__list -grid grid-cols-4"
      >
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/vasque-breeze-all-terrain-gtx-mens-hiking-boots-br/V7040W"
            onClick={[Function]}
          >
            <img
              className="-image"
              src="https://www.wildearth.com.au/assets/full/V7040W.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Vasque Breeze All-Terrain GTX Mens Hiking Boots - Brown Olive/Bossa Nova
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                369.99
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/vasque-breeze-all-terrain-gtx-womens-hiking-boots/V7023W"
            onClick={[Function]}
          >
            <img
              className="-image"
              src="https://www.wildearth.com.au/assets/full/V7023W.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Vasque Breeze All-Terrain GTX Womens Hiking Boots - Gargyle
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                369.99
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/scarpa-phantom-6000-mountaineering-boots-black-ora/SCA40017-BlackOrange"
            onClick={[Function]}
          >
            <img
              className="-image"
              src="https://www.wildearth.com.au/assets/full/SCA40017-BlackOrange.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Scarpa Phantom 6000 Unisex Waterproof Mountaineering Boots
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                1099.95
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/scarpa-mont-blanc-pro-gtx-goretex-mountaineering-b/SCA40013"
            onClick={[Function]}
          >
            <img
              className="-image"
              src="https://www.wildearth.com.au/assets/full/SCA40013.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Scarpa Mont Blanc Pro GTX Goretex Unisex Mountaineering Boots
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                749.95
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/scarpa-terra-gtx-unisex-shoes-brown/SCA00108"
            onClick={[Function]}
          >
            <img
              className="-image"
              src="https://www.wildearth.com.au/assets/full/SCA00108.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Scarpa Terra Goretex Unisex Waterproof Hiking Boots - Brown
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                349.95
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/scarpa-delta-leather-fly-mens-hiking-boots-mens-hi/SCA00090"
            onClick={[Function]}
          >
            <img
              className="-image"
              src="https://www.wildearth.com.au/assets/full/SCA00090.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Scarpa Delta Leather Fly Mens Waterproof Hiking Boots - T.Di Moro
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                399.95
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/scarpa-delta-womens-leather-waterproof-hiking-boot/SCA00065"
            onClick={[Function]}
          >
            <img
              className="-image"
              src="https://www.wildearth.com.au/assets/full/SCA00065.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Scarpa Delta Womens Goretex Waterproof Leather Hiking Boots
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                499.95
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/salomon-quest-prime-gtx-womens-hiking-boots-ebony/411299"
            onClick={[Function]}
          >
            <img
              className="-image"
              src="https://www.wildearth.com.au/assets/full/411299.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Salomon Quest Prime GTX Womens Hiking Boots - Ebony/Black/Icy Morn
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                299.99
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/salomon-outblast-ts-cswp-mens-hiking-boots-black-b/409223"
            onClick={[Function]}
          >
            <img
              className="-image"
              src="https://www.wildearth.com.au/assets/full/409223.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Salomon OUTblast TS CSWP Mens Hiking Boots - Black/Black/Black
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                259.99
                 
              </span>
            </div>
          </div>
        </div>
        <div
          className="UNX-productCard__container"
        >
          <a
            className="UNX-product-card UNX-grid-card"
            href="https://www.wildearth.com.au/buy/salomon-quest-prime-gtx-mens-hiking-shoes-phantom/404637"
            onClick={[Function]}
          >
            <img
              className="-image"
              src="https://www.wildearth.com.au/assets/full/404637.jpg"
            />
          </a>
          <div
            className="UNX-swatch__container"
          />
          <div
            className="-details"
          >
            <div
              className="-title"
            >
              Salomon Quest Prime Goretex Mens Waterproof Hiking Boots - Phantom/Black/Quiet Shade
            </div>
            <div
              className="-price"
            >
              <span
                className="-strike"
              >
                $
                299.99
                 
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div>
    <div
      className="UNX-searchbox__container"
    >
      <form
        onSubmit={[Function]}
      >
        <input
          className="UNX-searchbox__input"
          onChange={[Function]}
          placeholder=""
          value="boots"
        />
        <button
          className="UNX-searchbox__button"
          type="submit"
        >
          Search
        </button>
      </form>
    </div>
  </div>,
]
`;
